File: Allegro5 Test/background.cpp
#include "background.h"

background::background(){}

void background::Init(ALLEGRO_BITMAP *image, int x, int y)
{
	GameObject::Init(x, y,0,0,0,0,0,0);
	SetID(MISC);
	SetCollidable(false);

	background::image = image;
	frameWidth = al_get_bitmap_width(image);
}

void background::Destroy(){
	GameObject::Destroy();
}

void background::Update(){
	GameObject::Update();
}

void background::Render(){
	al_draw_bitmap(image, x, y, 0);
}

File: js/background.js
// Background class - static background elements

class Background extends GameObject {
    constructor() {
        super();
        this.setId(ID.MISC);
        this.setCollidable(false);
    }
    
    init(texture, x, y, width, height) {
        super.init(x, y, 0, 0, 0, 0, 0, 0);
        
        if (texture) {
            this.sprite = new PIXI.Sprite(texture);
            this.sprite.x = x;
            this.sprite.y = y;
            this.sprite.width = width;
            this.sprite.height = height;
        }
    }
    
    update() {
        // Background doesn't need to update
    }
    
    render() {
        super.render();
    }
}